pkg/block/s3/adapter.go

```diff
@@ -666,6 +666,7 @@ func (a *Adapter) CreateMultiPartUpload(ctx context.Context, obj block.ObjectPoi
 		Bucket:      aws.String(bucket),
 		Key:         aws.String(key),
 		ContentType: aws.String(""),
+		Expires:     aws.Time(time.Now().Add(a.preSignedExpiry)),
```

Comment:
Note however that this is not necessarily the expiration time that the generated URL will have! That is capped also by the STS expiry.